airbyte-integrations/bases/source-acceptance-test/unit_tests/test_spec.py

```diff
@@ -648,3 +649,105 @@ def test_additional_properties_is_true(connector_spec, expectation):
     t = _TestSpec()
     with expectation:
         t.test_additional_properties_is_true(connector_spec)
+
+
+@pytest.mark.parametrize(
+    "connector_spec, should_fail",
+    (
+        (
+            {
+                "connectionSpecification": {"type": "object", "properties": {"api_token": {"type": "string", "airbyte_secret": True}}}
+            },
+            False,
+        ),
+        (
+            {
+                "connectionSpecification": {"type": "object", "properties": {"api_token": {"type": "null"}}}
+            },
+            False,
```

Comment:
In this case you're logging a warning isn't it? Might be worth mocking the logger / using `caplog` to check a warning was logged.

Reply:
done